azurerm/internal/services/compute/linux_virtual_machine_resource.go

```diff
@@ -208,6 +208,12 @@ func resourceLinuxVirtualMachine() *schema.Resource {
 
 			"secret": linuxSecretSchema(),
 
+			"skip_shutdown": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
```

Comment:
since this field controls the behaviour - this wants to go [into the features block](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/internal/provider/features.go#L18-L21) instead of on the resource - can we move this there? That should then be available as `meta.(*clients.Client).Features.VirtualMachine.GracefulShutdown`

Reply:
This makes a lot of sense. I'm going to use `graceful_shutdown` as the argument name in the `virtual_machine` block.